### PR TITLE
Update to rails 6

### PR DESCRIPTION
**Pull request by JonathanHallam**
Bumps the version number to rails 6 and fixes a test that was failing because of this. Also uses the version of Jasmine that is currently on github. This has to be done because there are changes that are on git that have yet to be officially released.

### CODE REVIEW

**Comment by JonathanHallam** [comment]
@kevindew Kind of jumped the gun with the PR. Have just tried to update it and run it in docker which fails. I'll look into it now.

File: config/application.rb [hunk 1]
```diff
@@ -7,12 +7,13 @@
...
 require "action_controller/railtie"
-# require "action_mailer/railtie"
+require "action_mailer/railtie"
+require "action_mailbox/engine"
+require "action_text/engine"
 require "action_view/railtie"
 # require "action_cable/engine"
 require "sprockets/railtie"
 require "rails/test_unit/railtie"
-require "action_mailer/railtie"
```
Comment (ChrisBAshton) [inline config/application.rb]: Well done for consolidating the duplicate lines 10 and 15 👍

File: config/environments/production.rb [hunk 2]
```diff
@@ -22,23 +22,21 @@
   # Apache or NGINX already handles this.
   config.public_file_server.enabled = ENV["RAILS_SERVE_STATIC_FILES"].present?
 
-  # Compress JavaScripts and CSS.
+  # Compress CSS using a preprocessor.
```
Comment (ChrisBAshton) [inline config/environments/production.rb]: This comment is quite confusing as we've commented out the CSS preprocessor on line 27. 🤔

File: config/initializers/new_framework_defaults_6_0.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Be sure to restart your server when you modify this file.
```
Comment (ChrisBAshton) [inline config/initializers/new_framework_defaults_6_0.rb]: Worth adding in the commit message that you intend to edit this file in subsequent commits.

File: .rubocop.yml [hunk 4]
```diff
@@ -14,6 +14,7 @@ AllCops:
 
 Metrics/BlockLength:
   Exclude:
+    - 'config/environments/*.rb'
```
Comment (ChrisBAshton) [inline .rubocop.yml]: Nice.

File: config/initializers/new_framework_defaults_6_0.rb [hunk 5]
```diff
@@ -37,9 +37,9 @@
 # If you send mail in the background, job workers need to have a copy of
 # MailDeliveryJob to ensure all delivery jobs are processed properly.
 # Make sure your entire app is migrated and stable on 6.0 before using this setting.
-# Rails.application.config.action_mailer.delivery_job = "ActionMailer::MailDeliveryJob"
+Rails.application.config.action_mailer.delivery_job = "ActionMailer::MailDeliveryJob"
```
Comment (ChrisBAshton) [inline config/initializers/new_framework_defaults_6_0.rb]: Just reading the comment above this one, is it worth holding off on flipping this until after this PR is merged and deployed?

File: config/environments/production.rb (hunk 2 above)
Comment (JonathanHallam) [reply config/environments/production.rb]: yeah, fair play, this happened automatically when I upgraded. Will change it back :)

File: config/initializers/new_framework_defaults_6_0.rb (hunk 3 above)
Comment (JonathanHallam) [reply config/initializers/new_framework_defaults_6_0.rb]: I've put a note about this in the most recent commit. Do you want one in the commit that introduces this code as well?

File: config/initializers/new_framework_defaults_6_0.rb (hunk 5 above)
Comment (JonathanHallam) [reply config/initializers/new_framework_defaults_6_0.rb]: oops didn't spot this. Yes!

File: config/initializers/new_framework_defaults_6_0.rb (hunk 3 above)
Comment (ChrisBAshton) [reply config/initializers/new_framework_defaults_6_0.rb]: That's where I meant, yeah - sorry!

**Pull request merged**